### DISCUSSION TITLE
Sunbird Keycloak setup implementation

### DISCUSSION THREAD

**Discussion by shrutikaCG** [Sunbird-RC/community#288] [General]
Hi Team,

We are from Capgemini working for YuWaah project. We need help from sunbird team for the below points:
1. How to setup keycloak with sunbird RC and what are the prerequisites needed?
2. Is there any implementation for OTP based authentication in sunbird Keycloak for user sign up and login?
3. How to redirect back to application from browser?

**Comment by tejash-jl** [comment]
@shrutikaCG 
1. You can check the following registry cli https://docs.sunbirdrc.dev/developer-documentation/installation-guide. All the prerequisites are mentioned here. Using this cli you will be able to bring up all the services required. 
The following keycloak configurations need to be configured in SB-RC. https://github.com/Sunbird-RC/sunbird-rc-core/blob/main/docker-compose.yml#L36-L40
2. Yes all the reference implementations are using OTP based login.
3.  > How to redirect back to application from browser? 

 Can you share more context to this question

**Reply by shrutikaCG** [reply]
**How to redirect back to application from browser?** For this point when user login with Keycloak it will redirect to browser and there we will get access token but how we will come back to our application(native app) from browser with session token?
**Yes all the reference implementations are using OTP based login.** For this point does OTP based authentication support SMS OTP or mobile authenticator apps like Google Authenticator, Authy, etc.?

**Reply by tejash-jl** [reply]
@shrutikaCG 
2. Yes we have added SMS/email based OTP support.
3. You can pass `redirect_uri` in the query params to keycloak login page. You can check sunbird rc UI https://github.com/Sunbird-RC/sunbird-rc-ui.

**Reply by shrutikaCG** [reply]
For web application we can add redirect_uri in query params as you have said. But in our case it is **mobile app** so how we will redirect back to app from browser with the session token?

**Reply by shrutikaCG** [reply]
We also have following queries:
1. In our app we need to login and using mobile number and with SMS OTP(SMS provider service like Ozonetel). Is there any plugin available to cover this option with sunbird keycloak?
2. Can you please share the list of software we need to have before starting with sunbird RC?
3. There are 13 different repositories available in sunbird RC. All will be required to setup in our local system or not?

**Reply by tejash-jl** [reply]
@shrutikaCG 
1. For the issues, you're facing on the mobile app on redirection. You can check out the Keycloak libraries available for mobile apps. If you are using react native you can check this https://github.com/react-keycloak/react-native-keycloak. This issue is outside the sunbird-rc ecosystem.
2. You can check our docs for more detailed information on how to setup sunbird-rc locally, https://docs.sunbirdrc.dev/developer-documentation/installation-guide

**Comment by HarshiniCG** [comment]
1. What storage services sunbird or diksha is using on github community?
2. For registration are we using keycloak admin UI ?
